### PR TITLE
[netcore] Disable debugging code in lookup_intrins

### LINKED ISSUE #17457: [netcore] Slow debug code in lookup_intrins
There's leftover debug code in `lookup_intrins` method:

https://github.com/mono/mono/blob/9f50bdb2d944793d30ea2d052cdad96feb282003/mono/mini/simd-intrinsics-netcore.c#L87-L92

We should disable it at least for release builds, possible all builds except for some checked one.

/cc @vargaz

### PR DESCRIPTION
**Pull request by filipnavara**
Fixes #17457

### CODE REVIEW

File: mono/mini/simd-intrinsics-netcore.c [hunk 1]
```diff
@@ -84,13 +84,15 @@ lookup_intrins (guint16 *intrinsics, int size, MonoMethod *cmethod)
...
 
+#if FALSE
 	for (int i = 0; i < (size / sizeof (guint16)) - 1; ++i) {
 		if (method_name (intrinsics [i])[0] > method_name (intrinsics [i + 1])[0]) {
 			printf ("%s %s\n",method_name (intrinsics [i]), method_name (intrinsics [i + 1]));
 			g_assert_not_reached ();
 		}
 	}
-
+#endif
```
Comment (EgorBo) [inline mono/mini/simd-intrinsics-netcore.c]: Could you please leave a note about this in the list of intrinsics?

Comment (filipnavara) [reply mono/mini/simd-intrinsics-netcore.c]: I would but I just realized it is not sorted for BMI1/2. Not sure if it works as expected now.

Comment (filipnavara) [reply mono/mini/simd-intrinsics-netcore.c]: Nvm, it is sorted, just not in `simd-methods-netcore.h`

**Comment by vargaz** [comment]
@monojenkins build

**Comment by vargaz** [comment]
@monojenkins build failed

**Pull request merged**